crates/sui-core/src/authority.rs

```diff
@@ -1096,6 +1097,16 @@ impl AuthorityState {
             .tap_err(|_| {
                 self.update_overload_metrics("consensus");
             })?;
+
+        let pending_tx_count = self
+            .get_cache_commit()
+            .approximate_pending_transaction_count();
+        if pending_tx_count > self.config.execution_cache.backpressure_threshold_for_rpc() {
```

Comment:
I'm a little confused about how this works, it seems like it doesn't have anything to do with the BackpressureManager you added, or with checkpoint construction/execution?

is it deliberately two different backpressure points in the same PR?

Reply:
Yes, this is a totally different construction. I had to do it this way to make the simtest possible to write. If we use a unified backpressure mechanism here, then setting backpressure_threshold=1 causes the test to be unable to submit transactions reliably and it fails because of that, instead of testing what we care about (liveness of checkpoint construction).

Its possible we don't even need this but it seemed prudent - if somehow there is network-wide backpressure, we probably don't want to accept additional transactions